tests/EventListener/IndexEntityChangesTest.php

```diff
@@ -0,0 +1,118 @@
+use Symfony\Component\Messenger\MessageBusInterface;
+
+/**
+ * @covers \App\EventListener\IndexEntityChanges
+ */
+class IndexEntityChangesTest extends TestCase
+{
+    protected m\MockInterface $curriculumIndex;
+    protected m\MockInterface $learningMaterialIndex;
+
+    protected m\MockInterface $meshIndex;
+
+    protected m\MockInterface $usersIndex;
+
+    protected m\MockInterface $bus;
+
+    protected IndexEntityChanges $indexEntityChanges;
+
+    protected function setUp(): void
+    {
+        $this->curriculumIndex = m::mock(Curriculum::class);
+        $this->learningMaterialIndex = m::mock(LearningMaterials::class);
+        $this->meshIndex = m::mock(Mesh::class);
+        $this->usersIndex = m::mock(Users::class);
+        $this->bus = m::mock(MessageBusInterface::class);
+        $this->indexEntityChanges = new IndexEntityChanges(
+            $this->curriculumIndex,
+            $this->learningMaterialIndex,
+            $this->meshIndex,
+            $this->usersIndex,
+            $this->bus
+        );
+    }
+
+    protected function tearDown(): void
+    {
+        unset($this->curriculumIndex);
+        unset($this->learningMaterialIndex);
+        unset($this->meshIndex);
+        unset($this->usersIndex);
+        unset($this->bus);
+    }
+
+    public function testPostUpdateUserNoDispatchOnExaminedOnlyChange(): void
+    {
+        $objectManager = m::mock(EntityManagerInterface::class);
+        $unitOfWork = m::mock(UnitOfWork::class);
+        $entity = m::mock(UserInterface::class);
+        $changed = ['examined' => 'we only care about the key'];
+        $args = new PostUpdateEventArgs($entity, $objectManager);
+
+        $objectManager->shouldReceive('getUnitOfWork')->andReturn($unitOfWork);
+        $unitOfWork->shouldReceive('getEntityChangeSet')->with($entity)->andReturn($changed);
+
+        $this->indexEntityChanges->postUpdate($args);
+
+        $this->bus->shouldNotHaveReceived('dispatch');
+    }
+
+    public function testPostUpdateUserNoDispatchIfIndexerIsDisabled(): void
+    {
+        $objectManager = m::mock(EntityManagerInterface::class);
+        $unitOfWork = m::mock(UnitOfWork::class);
+        $entity = m::mock(UserInterface::class);
+        $changed = ['nyuk' => 'nyuk nuyk'];
+        $args = new PostUpdateEventArgs($entity, $objectManager);
+
+        $objectManager->shouldReceive('getUnitOfWork')->andReturn($unitOfWork);
+        $unitOfWork->shouldReceive('getEntityChangeSet')->with($entity)->andReturn($changed);
+        $this->usersIndex->shouldReceive('isEnabled')->andReturn(false);
+
+        $this->indexEntityChanges->postUpdate($args);
+
+        $this->bus->shouldNotHaveReceived('dispatch');
+    }
+
+    public function testPostUpdateUserDispatch(): void
+    {
+        $objectManager = m::mock(EntityManagerInterface::class);
+        $unitOfWork = m::mock(UnitOfWork::class);
+        $entity = m::mock(UserInterface::class);
+        $changed = ['nyuk' => 'nuyk nyuk'];
```

Comment:
Can you add `examined` as key here as well, just to make sure we don't ever switch to a `in_array` and not update when things are examined *and* changed?

Reply:
good point. handled. :heavy_check_mark: